velox/functions/prestosql/types/JsonCastOperator.cpp

```diff
@@ -1229,21 +1229,28 @@ void JsonCastOperator::castFromJson(
     maxSize = std::max(maxSize, input.size());
   });
   paddedInput_.resize(maxSize + simdjson::SIMDJSON_PADDING);
-  context.applyToSelectedNoThrow(rows, [&](auto row) {
-    writer.setOffset(row);
-    if (inputVector->isNullAt(row)) {
-      writer.commitNull();
-      return;
-    }
-    auto& input = inputVector->valueAt(row);
-    memcpy(paddedInput_.data(), input.data(), input.size());
-    simdjson::padded_string_view paddedInput(
-        paddedInput_.data(), input.size(), paddedInput_.size());
-    if (auto error = castFromJsonOneRow<kind>(paddedInput, writer)) {
-      context.setVeloxExceptionError(row, errors_[error]);
-      writer.commitNull();
-    }
-  });
+  context.applyToSelectedNoThrow(
+      rows,
+      [&](auto row) INLINE_LAMBDA {
+        writer.setOffset(row);
+        if (inputVector->isNullAt(row)) {
+          writer.commitNull();
+          return;
+        }
+        auto& input = inputVector->valueAt(row);
+        memcpy(paddedInput_.data(), input.data(), input.size());
+        simdjson::padded_string_view paddedInput(
+            paddedInput_.data(), input.size(), paddedInput_.size());
+        if (auto error = castFromJsonOneRow<kind>(paddedInput, writer)) {
+          context.setVeloxExceptionError(row, errors_[error]);
+          writer.commitNull();
+        }
+      },
+      [&](auto row) INLINE_LAMBDA {
+        context.setVeloxExceptionError(row, std::current_exception());
```

Comment:
error setting is taken care of by applyToSelectedNoThrow so you only need to commit the writer

https://github.com/facebookincubator/velox/blob/main/velox/expression/EvalCtx.h#L306-L309

Reply:
Fixed.